### PR TITLE
[INLONG-10247][Manager]  Support schedule information management for offline sync

### LINKED ISSUE #10247: [Feature][Manager] Support schedule information management for offline sync
### Description

As we want to support offline sync in InLong,  we should manage the schedule configurations.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aloyszhang**
Fixes #10247 

### Motivation

Add metadata management of schedule information for offline sync.

### Modifications

1. define key properties of  schedule info
2. support  CURD for schedule information 
3. support managing of schedule info using manager client

### Verifying this change

- [ ] This change added tests and can be verified as follows:

 - inlong-manager/manager-dao/src/test/java/org/apache/inlong/manager/dao/mapper/ScheduleEntityTest.java

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-manager/manager-dao/src/main/resources/mappers/ScheduleEntityMapper.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                #{crontabExpression, jdbcType=VARCHAR}, #{status,jdbcType=INTEGER},
+                #{previousStatus,jdbcType=INTEGER}, #{isDeleted,jdbcType=INTEGER},
+                #{creator,jdbcType=VARCHAR}, #{modifier,jdbcType=VARCHAR})
+    </insert>
+
+    <select id="selectByPrimaryKey" parameterType="java.lang.Integer" resultMap="BaseResultMap">
+        select
+        <include refid="Base_Column_List"/>
+        from schedule_config
+        where id = #{id,jdbcType=INTEGER}
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/ScheduleEntityMapper.xml:70]: and is_deleted = 0

File: inlong-manager/manager-dao/src/main/resources/mappers/ScheduleEntityMapper.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    </select>
+
+    <select id="selectByGroupId" parameterType="java.lang.String" resultMap="BaseResultMap">
+        select
+        <include refid="Base_Column_List"/>
+        from schedule_config
+        where group_id = #{groupId,jdbcType=VARCHAR}
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/ScheduleEntityMapper.xml:77]: and is_deleted = 0

File: inlong-manager/manager-dao/src/main/resources/mappers/ScheduleEntityMapper.xml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            <if test="taskParallelism != null">
+                task_parallelism = #{taskParallelism, jdbcType=INTEGER},
+            </if>
+            <if test="crontabExpression != null">
+                crontab_expression = #{crontabExpression, jdbcType=VARCHAR},
+            </if>
+        </set>
+        <where>
+            id = #{id, jdbcType=INTEGER}
+        </where>
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/ScheduleEntityMapper.xml:116]: and version = #{version,jdbcType=INTEGER}

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InLongScheduleApi.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+public interface InLongScheduleApi {
+
+    @POST("schedule/save")
+    Call<Response<Integer>> createSchedule(@Body ScheduleInfo scheduleInfo);
+
+    @GET("schedule/exist/{groupId}")
+    Call<Response<Boolean>> exist(@Path("groupId") String groupId);
+
+    @POST("schedule/update")
+    Call<Response<Boolean>> update(@Body ScheduleInfo scheduleInfo);
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InLongScheduleApi.java:40]: @Body ScheduleInfoRequest request

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InLongScheduleApi.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import retrofit2.http.DELETE;
+import retrofit2.http.GET;
+import retrofit2.http.POST;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+public interface InLongScheduleApi {
+
+    @POST("schedule/save")
+    Call<Response<Integer>> createSchedule(@Body ScheduleInfo scheduleInfo);
```
Comment (fuweng11) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InLongScheduleApi.java:34]: Call<Response<Integer>> createSchedule(@Body ScheduleInfoRequest request);

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleService.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+public interface ScheduleService {
+
+    /**
+     * Save schedule info.
+     *
+     * @param scheduleInfo schedule request need to save
+     * @param operator name of operator
+     * @return schedule info id in backend storage
+     */
+    int save(@Valid @NotNull(message = "schedule request cannot be null") ScheduleInfoRequest scheduleInfo,
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleService.java:35]: int save(@Valid @NotNull(message = "schedule request cannot be null") ScheduleInfoRequest request,

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleService.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    ScheduleInfo get(String groupId);
+
+    /**
+     * Modify schedule information
+     *
+     * @param scheduleInfo schedule request that needs to be modified
+     * @param operator name of operator
+     * @return whether succeed
+     */
+    Boolean update(@Valid @NotNull(message = "schedule request cannot be null") ScheduleInfoRequest scheduleInfo,
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleService.java:61]: Boolean update(@Valid @NotNull(message = "schedule request cannot be null") ScheduleInfoRequest request

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+    private static Logger LOGGER = LoggerFactory.getLogger(ScheduleServiceImpl.class);
+
+    @Autowired
+    private InlongGroupEntityMapper groupMapper;
+    @Autowired
+    private ScheduleEntityMapper scheduleEntityMapper;
+
+    @Override
+    public int save(ScheduleInfoRequest scheduleInfo, String operator) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleServiceImpl.java:48]: public int save(ScheduleInfoRequest requst, String operator)

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    @Override
+    public Boolean deleteByGroupId(String groupId, String operator) {
+        LOGGER.debug("begin to delete schedule info for groupId={}", groupId);
+        checkGroupExist(groupId);
+        ScheduleEntity entity = scheduleEntityMapper.selectByGroupId(groupId);
+        if (entity == null) {
+            LOGGER.error("schedule info for groupId {} does not exist", groupId);
+            return false;
+        }
+        int res = scheduleEntityMapper.deleteByGroupId(groupId);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleServiceImpl.java:100]: Please use logical deletion method.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    @Override
+    public Boolean update(ScheduleInfoRequest request, String operator) {
+        LOGGER.debug("begin to update schedule info={}", request);
+        String groupId = request.getGroupId();
+        ScheduleEntity entity = getScheduleEntity(groupId);
+        CommonBeanUtils.copyProperties(request, entity, true);
+        entity.setModifier(operator);
+        scheduleEntityMapper.updateByIdSelective(entity);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/schedule/ScheduleServiceImpl.java:86]: The update may fail and an exception needs to be thrown.

File: inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql [hunk 11]
```diff
@@ -957,4 +957,34 @@ CREATE TABLE IF NOT EXISTS `cluster_config`
...
+    `crontab_expression`     varchar(256) DEFAULT NULL COMMENT 'Crontab expression if schedule type is crontab',
+    `status`                 int(4)       DEFAULT '100' COMMENT 'Schedule status',
+    `previous_status`        int(4)       DEFAULT '100' COMMENT 'Previous schedule status',
+    `is_deleted`             int(11)      DEFAULT '0' COMMENT 'Whether to delete, 0: not deleted, > 0: deleted',
+    `creator`                varchar(64)  NOT NULL COMMENT 'Creator name',
+    `modifier`               varchar(64)  DEFAULT NULL COMMENT 'Modifier name',
+    `create_time`            timestamp    NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'Create time',
+    `modify_time`            timestamp    NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'Modify time',
+    `version`                int(11)      NOT NULL DEFAULT '1' COMMENT 'Version number, which will be incremented by 1 after modification',
+    PRIMARY KEY (`id`)
```
Comment (fuweng11) [inline inlong-manager/manager-test/src/main/resources/h2/apache_inlong_manager.sql:984]: Is (groupId, is_deleted) a unique key?

File: inlong-manager/manager-web/sql/apache_inlong_manager.sql [hunk 12]
```diff
@@ -1021,4 +1021,33 @@ CREATE TABLE IF NOT EXISTS `cluster_config`
...
+    `crontab_expression`     varchar(256) DEFAULT NULL COMMENT 'Crontab expression if schedule type is crontab',
+    `status`                 int(4)       DEFAULT '100' COMMENT 'Schedule status',
+    `previous_status`        int(4)       DEFAULT '100' COMMENT 'Previous schedule status',
+    `is_deleted`             int(11)      DEFAULT '0' COMMENT 'Whether to delete, 0: not deleted, > 0: deleted',
+    `creator`                varchar(64)  NOT NULL COMMENT 'Creator name',
+    `modifier`               varchar(64)  DEFAULT NULL COMMENT 'Modifier name',
+    `create_time`            timestamp    NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'Create time',
+    `modify_time`            timestamp    NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'Modify time',
+    `version`                int(11)      NOT NULL DEFAULT '1' COMMENT 'Version number, which will be incremented by 1 after modification',
+    PRIMARY KEY (`id`)
```
Comment (fuweng11) [inline inlong-manager/manager-web/sql/apache_inlong_manager.sql:1048]: Ditto.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InLongSchedulerController.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+    @ApiOperation(value = "Save schedule info")
+    public Response<Integer> save(@RequestBody ScheduleInfoRequest request) {
+        int result = scheduleService.save(request, LoginUserUtils.getLoginUser().getName());
+        return Response.success(result);
+    }
+
+    @RequestMapping(value = "/schedule/exist/{groupId}", method = RequestMethod.GET)
+    @ApiOperation(value = "Is the schedule info exists for inlong group")
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "groupId", dataTypeClass = String.class, required = true)
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InLongSchedulerController.java:62]: @ApiImplicitParam(name = "groupId", dataTypeClass = String.class, required = true)

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InLongSchedulerController.java [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    })
+    public Response<Boolean> exist(@PathVariable String groupId) {
+        return Response.success(scheduleService.exist(groupId));
+    }
+
+    @RequestMapping(value = "/schedule/get", method = RequestMethod.GET)
+    @ApiOperation(value = "Get schedule info for inlong group")
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "groupId", dataTypeClass = String.class, required = true)
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InLongSchedulerController.java:71]: @ApiImplicitParam(name = "groupId", dataTypeClass = String.class, required = true)

**Comment by aloyszhang** [comment]
@fuweng11 I resolved the comments, PTAL again

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/schedule/ScheduleInfo.java [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+@ApiModel("Schedule response")
+public class ScheduleInfo {
+
+    @ApiModelProperty(value = "Primary key")
+    @NotNull(groups = UpdateValidation.class)
+    private Integer id;
+
+    @ApiModelProperty("Inlong Group ID")
+    @NotNull
+    private String groupId;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/schedule/ScheduleInfo.java:46]: inlongGroupId

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**